tests/postgres_test.py

```diff
@@ -15,7 +26,41 @@ def test_get_columns_from_table():
 
     expectation = ["id", "answer", "question"]
 
-    columns = PostgresConnector(**CREDENTIALS).get_columns_from_table(
+    columns = PostgresConnector(CREDENTIALS).get_columns_from_table(
         target_schema="public", target_table="test"
     )
     assert columns == expectation
+
+
+@pytest.mark.parametrize(
+    "test_name, schema, table, column_name, result",
+    [
+        (
+            "unique",
+            "schema",
+            "table",
+            "column",
+            [
+                call(
+                    """select count(*) as errors from(
+                select column from schema.table where column is not null group by column having count(*) > 1 )
+                errors"""
+                )
+            ],
+        ),
+        (
+            "not_null",
+            "schema",
+            "table",
+            "column",
+            [call("select count(*) as errors from schema.table where column is null")],
+        ),
+    ],
+)
+def test_run_test(mocker, test_name, schema, table, column_name, result):
+    execute_and_check = mocker.patch(
+        "dbt_sugar.core.connectors.postgres_connector.PostgresConnector.execute_and_check"
+    )
+    postgres_connector = PostgresConnector(CREDENTIALS)
+    postgres_connector.run_test(test_name, schema, table, column_name)
+    execute_and_check.assert_has_calls(result)
```

Comment:
Can we not do an actual test? We do spring up a postgres docker so I think we can go all the way here don't you think?